test/jdk/java/nio/channels/vthread/BlockingChannelOps.java

```diff
@@ -186,15 +196,20 @@ public void testSocketChannelWriteAsyncClose() throws Exception {
         VThreadRunner.run(() -> {
             try (var connection = new Connection()) {
                 SocketChannel sc = connection.channel1();
-                ScheduledCloser.schedule(sc, DELAY);
+                runAfterParkedAsync(sc::close);
                 try {
                     ByteBuffer bb = ByteBuffer.allocate(100*1024);
                     for (;;) {
                         int n = sc.write(bb);
                         assertTrue(n > 0);
                         bb.clear();
                     }
```

Comment:
Hello Alan, since `AsynchronousCloseException` is expected here, should we add a `fail` after this loop if we don't receive that exception

Reply:
It's an infinite loop as write may need to be called many times before it blocks. So you can't put a `fail` after the loop as it will never be executed. If another exception is thrown then the test will fail.

Reply:
You are right; I overlooked the part that this `for` loop is infinite. So, for this test method to complete, the only way out of that `for` loop is some exception getting thrown and we then catch the expected exception and ignore it or let the unexpected exception propagate into a test failure. So yes, the current code looks fine.